src/main/scala/firrtl/passes/RemoveIntervals.scala

```diff
@@ -48,29 +50,32 @@ class RemoveIntervals extends Pass with PreservesAll[Transform] {
   def run(c: Circuit): Circuit = {
     val alignedCircuit = c
     val errors = new Errors()
-    val wiredCircuit = alignedCircuit map makeWireModule
-    val replacedCircuit = wiredCircuit map replaceModuleInterval(errors)
+    val replacedNodes = mutable.HashMap.empty[OfModule, mutable.HashSet[String]]
+    val wiredCircuit = alignedCircuit map makeWireModule(replacedNodes)
+    val replacedCircuit = wiredCircuit map replaceModuleInterval(errors, replacedNodes)
     errors.trigger()
     InferTypes.run(replacedCircuit)
   }
 
   /* Replace interval types */
-  private def replaceModuleInterval(errors: Errors)(m: DefModule): DefModule =
-    m map replaceStmtInterval(errors, m.name) map replacePortInterval
+  private def replaceModuleInterval(errors: Errors, replacedNodes: mutable.HashMap[OfModule, mutable.HashSet[String]])(m: DefModule): DefModule =
+    m map replaceStmtInterval(errors, m.name, replacedNodes(OfModule(m.name)).toSet) map replacePortInterval
 
-  private def replaceStmtInterval(errors: Errors, mname: String)(s: Statement): Statement = {
+  private def replaceStmtInterval(errors: Errors, mname: String, replacedNodes: Set[String])(s: Statement): Statement = {
     val info = s match {
       case h: HasInfo => h.info
       case _ => NoInfo
     }
-    s map replaceTypeInterval map replaceStmtInterval(errors, mname) map replaceExprInterval(errors, info, mname)
+    s map replaceTypeInterval map replaceStmtInterval(errors, mname, replacedNodes) map replaceExprInterval(errors, info, mname, replacedNodes)
 
   }
 
-  private def replaceExprInterval(errors: Errors, info: Info, mname: String)(e: Expression): Expression = e match {
+  private def replaceExprInterval(errors: Errors, info: Info, mname: String, replacedNodes: Set[String])(e: Expression): Expression = e match {
+    case ref: WRef if replacedNodes(ref.name) =>
+      ref.copy(kind = WireKind)
     case _: WRef | _: WSubIndex | _: WSubField => e
```

Comment:
I see you're fixing the Kind, are the types correct? I see there is `replaceTypeInterval`, but as far as I can tell, this is not applied to `Expressions`.

Reply:
yeah It's used for ports, but it doesn't look like the types fixedup are afterwards

Reply:
I'm wondering if instead we should just invalidate types and kinds. Doing the kinds here is faster than running `ResolveKinds`, but if we have to run `InferTypes` anyway 🤷

Reply:
oh wait my bad. it does run `InferTypes` afterwards https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/passes/RemoveIntervals.scala#L54. so I guess just delete that line and invalidate `InferTypes` instead?

Reply:
Well unfortunately this mixes in `PreservesAll[Transform]` so we can't change that behavior in a backportable way. Perhaps the best approach is to do what you originally did, leave the `InferTypes` call alone (change it in a non-backportable 1.4 version of this PR) and just fixup the Kinds as you did.

Reply:
@seldridge thoughts?

Reply:
There should be two PRs or backport this and change it in 1.3.x:

1) 1.4: Remove `PreservesAll[Transform]` and add the appropriate invalidations
2) 1.3.x: Manually run the invalidated transforms at the finish of this transform

The lack of backportability of `PreservesAll` was my bad. This is intended to be a "helper" mix-in, but it's just busted. Sorry.